src/System/include/BipedalLocomotion/System/ILinearTaskSolver.h

```diff
@@ -48,14 +51,30 @@ template <class _Task, class _State> class ILinearTaskSolver : public Source<_St
      * @param taskName unique name associated to the task.
      * @param priority Priority associated to the task. The lower the number the higher the
      * priority.
-     * @param weight Weight associated to the task. This parameter is optional. The default value is
-     * an object that does not contain any value. So is an invalid weight.
-     * @return true if the task has been added to the inverse kinematics.
+     * @param weightProvider Weight provider associated to the task. This parameter is optional. The
+     * default value is an object that does not contain any value. So is an invalid provider.
```

Comment:
As for ``QPTSID``, maybe it is worth explaining a bit what it means to set an invalid provider